model/riscv_sys_control.sail

```diff
@@ -408,8 +482,187 @@ function handle_exception(e: ExceptionType) -> unit = {
+          tdata2_x.tdata2[i][(xlen - 1) .. idx] == tdata2_content[(xlen - 1) .. idx]
+        },
+        MOPT_GE    => { (tdata2_content >=_u tdata2_x.tdata2[i][(xlen - 1) .. 0]) },
+        MOPT_LT    => { (tdata2_content <_u  tdata2_x.tdata2[i][(xlen - 1) .. 0]) },
+        MOPT_MASK_LOW => {
+          let intemrd_cmpr_val : half_xlenbits = tdata2_content[(half_xlen - 1) .. 0] & tdata2_x.tdata2[i][(xlen - 1) .. half_xlen];
+          intemrd_cmpr_val == tdata2_x.tdata2[i][(half_xlen - 1) .. 0]
+        },
+        MOPT_MASK_HIGH => {
+          let intemrd_cmpr_val : half_xlenbits = tdata2_content[(xlen - 1) .. half_xlen] & tdata2_x.tdata2[i][(xlen - 1) .. half_xlen];
+          intemrd_cmpr_val == tdata2_x.tdata2[i][(half_xlen - 1) .. 0]
+        },
+        MOPT_NOT_EQUAL => { (tdata2_x.tdata2[i] != tdata2_content) },
+        MOPT_NOT_NAPOT => {
+          let idx : {'n, (1 <= 'n < xlen). int('n)} = find_first_zero(tdata2_x.tdata2[i]);
+          tdata2_x.tdata2[i][(xlen - 1) .. idx] != tdata2_content[(xlen - 1) .. idx]
+        },
+        MOPT_NOT_MASK_LOW => {
+          let intemrd_cmpr_val : half_xlenbits = tdata2_content[(half_xlen - 1) .. 0] & tdata2_x.tdata2[i][(xlen - 1) .. half_xlen];
+          intemrd_cmpr_val != tdata2_x.tdata2[i][(half_xlen - 1) .. 0]
+        },
+        MOPT_NOT_MASK_HIGH => {
+          let intemrd_cmpr_val : half_xlenbits = tdata2_content[(xlen - 1) .. half_xlen] & tdata2_x.tdata2[i][(xlen - 1) .. half_xlen];
+          intemrd_cmpr_val != tdata2_x.tdata2[i][(half_xlen - 1) .. 0]
+        },
+        _          => false
+      };
+      if ((((MCtrl6[m] == 0b1) & (cur_priv == Machine))     | /* Trigger is enabled in M-mode */
+           ((MCtrl6[s] == 0b1) & (cur_priv == Supervisor))  | /* Trigger is enabled in S-mode */
+           ((MCtrl6[u] == 0b1) & (cur_priv == User)) )      & /* Trigger is enabled in U-mode */
+           (TriggerFireMatch(cur_priv))) then { /* Trigger match valid? check reentranacy */
+            match match_type {
+              /* Check For PC or Instruction Match */
+              INSTR_MATCH => {
+                if (data_matched & size_matched & (MCtrl6[execute] == 0b1)) then {
+                  MCtrl6[hit0] = 0b1;
+                  tdata1_x.tdata1[i][26 .. 0] = MCtrl6.bits;
+                  raise_exception = true;
+                }
+              },
+              /* Check for Load Address match */
+              LOAD_MATCH_BEFORE => {
+                if (data_matched & size_matched & (MCtrl6[load] == 0b1)) then {
+                    MCtrl6[hit0] = 0b1;
+                    tdata1_x.tdata1[i][26 .. 0] = MCtrl6.bits;
+                    raise_exception = true;
+                }
+              },
+              /* Check for load data match */
+              LOAD_MATCH_AFTER => {
+                if (data_matched & size_matched & (MCtrl6[load] == 0b1)) then {
+                    MCtrl6[hit1] = 0b1;
```

Comment:
Can you set hit0 here as well? I assume in sail triggers are precise and occur immediately after the instruction that caused them to fire.

Reply:
Done